hooks/getContactDetails.js

```diff
@@ -0,0 +1,130 @@
+import {
+  materialExternalLink,
+  materialFacebook,
+  materialInstagram,
+  materialTwitter,
+} from "@/assets/Icons8";
+import { Fragment } from "react";
+
+export function getFullName(contact) {
+  const { firstName, middleName, lastName } = contact;
+
+  let displayName = "";
+
+  if (firstName) {
+    displayName += firstName;
+  }
+  if (middleName) {
+    displayName += " " + middleName;
+  }
+  if (lastName) {
+    displayName += " " + lastName;
+  }
+
+  return displayName.trim();
+}
+
+export function getProfilePicture(contact) {
+  const path = "/profile-pictures/";
+
+  const { profilePicture } = contact;
+
+  if (profilePicture) {
+    return path + profilePicture;
+  }
+}
+
+export function getSocialMediaIcon(platform) {
+  if (platform === "Instagram") {
+    return materialInstagram;
+  }
+
+  if (platform === "Twitter") {
+    return materialTwitter;
+  }
+
+  if (platform === "Facebook") {
+    return materialFacebook;
+  }
+
+  return materialExternalLink;
+}
+
+export function getSocialMediaHyperlink(platform, username) {
+  if (platform === "Instagram") {
+    return `https://www.instagram.com/${username}/`;
+  }
+  if (platform === "Twitter") {
+    return `https://twitter.com/${username}`;
+  }
+  if (platform === "Facebook") {
+    return `https://www.facebook.com/${username}`;
+  }
+  return null;
+}
+
+export function getGoogleMapsLink(address) {
+  const { street, zipCode, city, country } = address;
+  let encodedAddress = "";
+
+  if (street) {
+    encodedAddress += encodeURIComponent(street);
+  }
+
+  if (zipCode) {
+    if (encodedAddress) {
+      encodedAddress += ", ";
+    }
+    encodedAddress += encodeURIComponent(zipCode);
+  }
+
+  if (city) {
+    if (encodedAddress) {
+      encodedAddress += ", ";
+    }
+    encodedAddress += encodeURIComponent(city);
+  }
+
+  if (country) {
+    if (encodedAddress) {
+      encodedAddress += ", ";
+    }
+    encodedAddress += encodeURIComponent(country);
+  }
+
+  const googleMapsLink = `https://www.google.com/maps/search/?api=1&query=${encodedAddress}`;
+
+  return googleMapsLink;
+}
+
+export function getAddress(address) {
```

Comment:
Here i'd argue that it would be simpler to create a React component

```jsx
export function Address({ street, zipCode, city, country }) {
  return (
    <>
      {street && (
        <>
          {street}
          <br />
        </>
      )}
      {(zipCode || city) && (
        <>
          {zipCode} {city}
          <br />
        </>
      )}
      {country && <>{country}</>}
    </>
  );
}
```

of course this would go into your component folder. :)

Reply:
That should not be a problem:
```js
value={<Address {...addr} />}
```

Reply:
Thx, that works.